tools/walletextension/test/apis.go

```diff
@@ -4,87 +4,134 @@ import (
+	return api.reEncryptParams(encryptedParams)
 }
 
-func (api *DummyEthAPI) GetBalance(context.Context, common.EncryptedParamsGetBalance) (string, error) {
-	return api.encryptedSuccess()
+func (api *DummyAPI) GetBalance(_ context.Context, encryptedParams common.EncryptedParamsGetBalance) (string, error) {
+	return api.reEncryptParams(encryptedParams)
 }
 
-func (api *DummyEthAPI) GetTransactionByHash(context.Context, common.EncryptedParamsGetTxByHash) (*string, error) {
-	encryptedSuccess, err := api.encryptedSuccess()
-	return &encryptedSuccess, err
+func (api *DummyAPI) GetTransactionByHash(_ context.Context, encryptedParams common.EncryptedParamsGetTxByHash) (*string, error) {
+	reEncryptParams, err := api.reEncryptParams(encryptedParams)
+	return &reEncryptParams, err
 }
 
-func (api *DummyEthAPI) GetTransactionCount(context.Context, common.EncryptedParamsGetTxCount) (string, error) {
-	return api.encryptedSuccess()
+func (api *DummyAPI) GetTransactionCount(_ context.Context, encryptedParams common.EncryptedParamsGetTxCount) (string, error) {
+	return api.reEncryptParams(encryptedParams)
 }
 
-func (api *DummyEthAPI) GetTransactionReceipt(context.Context, common.EncryptedParamsGetTxReceipt) (*string, error) {
-	encryptedSuccess, err := api.encryptedSuccess()
-	return &encryptedSuccess, err
+func (api *DummyAPI) GetTransactionReceipt(_ context.Context, encryptedParams common.EncryptedParamsGetTxReceipt) (*string, error) {
+	reEncryptParams, err := api.reEncryptParams(encryptedParams)
+	return &reEncryptParams, err
 }
 
-func (api *DummyEthAPI) SendRawTransaction(context.Context, common.EncryptedParamsSendRawTx) (string, error) {
-	return api.encryptedSuccess()
+func (api *DummyAPI) SendRawTransaction(_ context.Context, encryptedParams common.EncryptedParamsSendRawTx) (string, error) {
+	return api.reEncryptParams(encryptedParams)
 }
 
-func (api *DummyEthAPI) EstimateGas(context.Context, common.EncryptedParamsEstimateGas, *rpc.BlockNumberOrHash) (*string, error) {
-	encryptedSuccess, err := api.encryptedSuccess()
-	return &encryptedSuccess, err
+func (api *DummyAPI) EstimateGas(_ context.Context, encryptedParams common.EncryptedParamsEstimateGas, _ *rpc.BlockNumberOrHash) (*string, error) {
+	reEncryptParams, err := api.reEncryptParams(encryptedParams)
+	return &reEncryptParams, err
 }
 
-// Returns the message `successMsg`, encrypted with the viewing key set via `setViewingKey`.
-func (api *DummyEthAPI) encryptedSuccess() (string, error) {
-	encryptedBytes, err := ecies.Encrypt(rand.Reader, api.viewingKey, []byte(successMsg), nil, nil)
+func (api *DummyAPI) Logs(ctx context.Context, encryptedParams common.EncryptedParamsLogSubscription) (*rpc.Subscription, error) {
+	params, err := api.enclavePrivateKey.Decrypt(encryptedParams, nil, nil)
+	if err != nil {
+		return nil, fmt.Errorf("could not decrypt params with enclave private key")
+	}
+
+	notifier, _ := rpc.NotifierFromContext(ctx)
+	sub := notifier.CreateSubscription()
+	go func() {
```

Comment:
I don't think this ever dies?

Reply:
It'll be reaped once the test ends. I think that's fine here.

Reply:
The Go test engine cleans up go routines after every test?